src/main/java/org/codehaus/mojo/flatten/FlattenMode.java

```diff
@@ -114,6 +117,32 @@ public FlattenDescriptor getDescriptor()
             case fatjar:
                 descriptor.setDependencies( ElementHandling.remove );
                 break;
+            case versionOnly:
+                descriptor.setBuild( ElementHandling.keep );
+                descriptor.setCiManagement( ElementHandling.keep );
+                descriptor.setContributors( ElementHandling.keep );
+                descriptor.setDependencies( ElementHandling.keep );
+                descriptor.setDependencyManagement( ElementHandling.keep );
+                descriptor.setDescription( ElementHandling.keep );
+                descriptor.setDevelopers( ElementHandling.keep );
+                descriptor.setDistributionManagement( ElementHandling.keep );
+                descriptor.setInceptionYear( ElementHandling.keep );
+                descriptor.setIssueManagement( ElementHandling.keep );
+                descriptor.setMailingLists( ElementHandling.keep );
+                descriptor.setModules( ElementHandling.keep );
+                descriptor.setName( ElementHandling.keep );
+                descriptor.setOrganization( ElementHandling.keep );
+                descriptor.setParent( ElementHandling.keep );
+                descriptor.setPluginManagement( ElementHandling.keep );
+                descriptor.setPluginRepositories( ElementHandling.keep );
+                descriptor.setPrerequisites( ElementHandling.keep );
+                descriptor.setProfiles( ElementHandling.keep );
+                descriptor.setProperties( ElementHandling.keep );
+                descriptor.setReporting( ElementHandling.keep );
+                descriptor.setRepositories( ElementHandling.keep );
+                descriptor.setScm( ElementHandling.keep );
+                descriptor.setUrl( ElementHandling.keep );
+                descriptor.setVersion( ElementHandling.expand );

```

Comment:
adding a break here would be more clean than an uncommented fallthrough.

Reply:
Will do.